source/FreeRTOS_DNS_Parser.c

```diff
@@ -735,10 +743,12 @@
                                          &( pxSet->pucByte[ sizeof( DNSAnswerRecord_t ) ] ),
                                          ipSIZE_OF_IPv6_ADDRESS );
 
-                        if( ppxAddressInfo != NULL )
-                        {
-                            pxNewAddress = pxNew_AddrInfo( pxSet->pcName, FREERTOS_AF_INET6, xIP_Address.xIPAddress.xIP_IPv6.ucBytes );
-                        }
+                        #if ( ipconfigUSE_DNS_CACHE == 1 ) || ( ipconfigDNS_USE_CALLBACKS == 1 )
```

Comment:
Could you add some more comments in the code to help developers understand why wrapping this is still working if pxNewAddress is NULL when CACHE/CALLBACK is disabled.